cinema/tests/test_view.py

```diff
@@ -0,0 +1,127 @@
+    movie_with_genres.genres.add(genre1, genre2)
+
+
+def detail_url(pk):
+    return reverse("cinema:movie-detail", args=[pk])
+
+
+class TestMovieUnauthenticated(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class TestMovieUserAuthenticated(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            "test@test.com",
+            "testpass"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_list_movies(self):
+        create_test_movies()
+
+        res = self.client.get(MOVIE_URL)
+
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_movies_with_filters(self):
+        create_test_movies()
+        sample_movie(title="Hello, World!")
+
+        res = self.client.get(MOVIE_URL, {"title": "Test"})
+
+        serializer1 = MovieListSerializer(Movie.objects.get(id=1), many=False)
+        serializer2 = MovieListSerializer(Movie.objects.get(id=2), many=False)
+        serializer3 = MovieListSerializer(Movie.objects.get(id=3), many=False)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertIn(serializer1.data, res.data)
+        self.assertIn(serializer2.data, res.data)
+        self.assertNotIn(serializer3.data, res.data)
+
+        res = self.client.get(MOVIE_URL, {"genres": [1, 2]})
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertIn(serializer1.data, res.data)
+        self.assertNotIn(serializer2.data, res.data)
+        self.assertNotIn(serializer3.data, res.data)
+
+        res = self.client.get(MOVIE_URL, {"actors": [1, 2]})
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertNotIn(serializer1.data, res.data)
+        self.assertIn(serializer2.data, res.data)
+        self.assertNotIn(serializer3.data, res.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.genres.add(Genre.objects.create(name="Test"))
+
+        url = detail_url(movie.id)
+        res = self.client.get(url)
+
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "Movie",
+            "description": "Description",
+            "duration": 10,
+        }
+
+        res = self.client.post(MOVIE_URL, payload)
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+class MovieImageUploadTests(TestCase):
+    pass
```

Comment:
Add tests here

Reply:
there are same tests in other file, so I deleted it from here